docs/pallas/tpu/pipelining.md

```diff
@@ -65,384 +76,117 @@ Let's talk about the components of this diagram in more detail:
   Values can be loaded into memory from their respective caches (VMEM for
   VREGs and SMEM for SREGs).
 * **Compute units**: A TensorCore has a scalar unit, vector unit (VPU) and
-  matrix unit (MXU) that can do numerical computation.
+  matrix unit (MXU) that can do numerical computation. Each of these compute units can operate asynchronously, but this is managed by the TPU compiler and thus from the programmer's perspective a TPU program is single-threaded.
   Compute units operate on values that live in SREGs and VREGs and output
   values into those registers as well.
 
-In order to do a vectorized computation on our values `x` and `y` that live
-in HBM, we need to:
-
-1. Copy the values `x` and `y` into VMEM.
-2. Load the values from VMEM into VREGs.
-3. Execute the computation using the VPU or MXU, storing the output in VREGs.
-4. Store the values in the output VREGs into VMEM.
-5. Copy the output values in VMEM back to HBM.
 
-+++ {"id": "TzctMbNsn3vc"}
++++ {"id": "8Tl3wt5Wk3Ek"}
 
-Let's implement a Pallas function that does just that!
+## TPU-specific Pipelining Features
 
-```{code-cell}
-:id: 2IXQxNWrKJyb
-:outputId: d62eb493-5f92-4496-f113-d3cd24cb0b9f
-
-def add_matrices_kernel(x_vmem_ref, y_vmem_ref, z_vmem_ref):
-  # Load x and y from VMEM into VREGs
-  x_vregs = x_vmem_ref[:, :]
-  y_vregs = y_vmem_ref[:, :]
-  # Execute a vectorized add
-  z_vregs = x_vregs + y_vregs
-  # Store the output values in VREGs back into VMEM
-  z_vmem_ref[:, :] = z_vregs
+Pallas TPU supports the following platform-specific features.
 
++++ {"id": "1jg5WmExk47l"}
 
-def add_matrices(x: jax.Array, y: jax.Array) -> jax.Array:
-  # pallas_call will first allocate scratch buffers for `x` and `y` in VMEM.
-  # It will then copy `x` and `y` from HBM into VMEM.
-  z = pl.pallas_call(
-      add_matrices_kernel, out_shape=jax.ShapeDtypeStruct(x.shape, x.dtype)
-  )(x, y)
-  # pallas_call will also copy the output from VMEM back into HBM.
-  return z
 
+### TPU Memory Spaces
```

Comment:
This looks like a more general TPU documentation that might be best to put elsewhere. Perhaps relate the memory spaces accessible from Pallas to the diagram we have in the basic pipelining docs?

Reply:
Hmm I agree that it's not pipelining specific, but this probably needs to be covered somewhere in the pipelining docs. I added your suggestion of mapping the TPU memory space to the common memory name (DRAM/SRAM).